gdx/src/com/badlogic/gdx/graphics/g3d/particles/influencers/DynamicsModifier.java

```diff
@@ -432,6 +432,133 @@ public TangentialAcceleration copy () {
 		}
 	}
 
+	public static class CircularMotion extends Angular {
+		FloatChannel positionChannel;
+		FloatChannel previousPositionChannel;
+		FloatChannel accelerationChannel;
+
+		public CircularMotion () {
+		}
+
+		public CircularMotion (CircularMotion motion) {
+			super(motion);
+		}
+
+		@Override
+		public void allocateChannels () {
+			super.allocateChannels();
+			positionChannel = controller.particles.addChannel(ParticleChannels.Position);
+			previousPositionChannel = controller.particles.addChannel(ParticleChannels.PreviousPosition);
+			accelerationChannel = controller.particles.addChannel(ParticleChannels.Acceleration);
+		}
+
+		@Override
+		public void activateParticles (int startIndex, int count) {
+			super.activateParticles(startIndex, count);
+			for (int i = startIndex; i < startIndex + count; ++i) {
+				int s = i * strengthChannel.strideSize;
+				int l = i * lifeChannel.strideSize;
+				int a = i * angularChannel.strideSize;
+				int posI = positionChannel.strideSize * i;
+				int prevPosI = previousPositionChannel.strideSize * i;
+
+				float lifePercent = lifeChannel.data[ParticleChannels.LifePercentOffset + l];
+				float speed = strengthChannel.data[s + ParticleChannels.VelocityStrengthStartOffset]
+					+ strengthChannel.data[s + ParticleChannels.VelocityStrengthDiffOffset] * strengthValue.getScale(lifePercent);
+				float phi = angularChannel.data[a + ParticleChannels.VelocityPhiStartOffset]
+					+ angularChannel.data[a + ParticleChannels.VelocityPhiDiffOffset] * phiValue.getScale(lifePercent);
+				float theta = angularChannel.data[a + ParticleChannels.VelocityThetaStartOffset]
+					+ angularChannel.data[a + ParticleChannels.VelocityThetaDiffOffset] * thetaValue.getScale(lifePercent);
+				float cosTheta = MathUtils.cosDeg(theta);
+				float sinTheta = MathUtils.sinDeg(theta);
+				float cosPhi = MathUtils.cosDeg(phi);
+				float sinPhi = MathUtils.sinDeg(phi);
+
+				TMP_V1.set(positionChannel.data[posI + ParticleChannels.XOffset], positionChannel.data[posI
+					+ ParticleChannels.YOffset], positionChannel.data[posI + ParticleChannels.ZOffset]);
+				TMP_V3.set(cosTheta * sinPhi, cosPhi, sinTheta * sinPhi);
+
+				if (!isGlobal) {
+					TMP_V3.mul(controller.transform.getRotation(TMP_Q, true));
+					TMP_V1.sub(controller.transform.getTranslation(TMP_V5));
+				}
+				// TMP_V1 is position
+				// TMP_V3 is axis of rotation
+				TMP_V4.set(TMP_V3).crs(TMP_V1); // TMP_V4 is now the direction of velocity (axis cross position)
+				TMP_V5.set(TMP_V3).scl(TMP_V1.dot(TMP_V3)); // TMP_V5 is now the projection of position onto the axis
+				TMP_V6.set(TMP_V5).sub(TMP_V1); // TMP_V6 is now the vector from the particle to the closest point on the axis
```

Comment:
If it's preferred, the vector from the particle to the axis can also be computed as 
`TMP_V6.set(TMP_V3).crs(TMP_V4)` 
(in other words, axis cross velocity is the direction of the force)